src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -80,16 +80,20 @@ update_shell(unsigned int index, const std::string& primary, const std::string&
   auto pri = std::make_unique<firmwareImage>(primary.c_str(), MCS_FIRMWARE_PRIMARY);
   if (pri->fail())
     throw xrt_core::error(boost::str(boost::format("Failed to read %s") % primary));
-
+  
+  auto stripped = std::make_unique<firmwareImage>(primary.c_str(), STRIPPED_FIRMWARE);
```

Comment:
firmwareImage ctor should be changed to take `const std::string&` instead of `const char*`.  The constructor is already converting the const char* to a std::string so might as well take std::string as argument.